docs/designs/ruby_test_plan_002.md

```diff
@@ -0,0 +1,112 @@
+       Given  clean repository of a "well-defined project"
+        When  a user request to fetch dependencies for a "well-defined project"
+        Then  direct and transitive dependencies are fetched
+         And  dependencies could be found within the "well-defined project"
+    Examples:
+        |    well-defined project                                       |
+        | Has a mixture of gems, Git and path dependencies and .gemspec |
+        | Has a mixture of gems, Git and path dependencies              |
+
+    Scenario outline: a user wants to fetch dependencies for a package containing ill-defined Gemfile.lock
+       Given  clean repository of a "ill-defined project"
+        When  a user request to fetch dependencies for a "ill-defined project"
+        Then  Cachi2 reports an error
+         And  repository for "ill-defined project" remains unchanged
+    Examples:
+        |    ill-defined project                     |
+        | Gemfile.lock is missing                    |
+        | Git dependency is not pinned to hash       |
+        | Transitive dependency is ill-defined       |
+
+
+@integration_test
+Feature: Cachi2 can preset configuration for hermetic builds
+
+    Scenario outline: a user prepares a repository with a Ruby project
+        When  a user generates environment for "a Ruby project"
+        Then  the environment contains a pointer to local Bundler config
+         And  Bundler cache directory structure is recreated for every project
+    Examples:
+        |    a Ruby project                                       |
+        | Single-project repository with mixed depenendcies types |
+        | Multi-project repository with mixed depenendcies types  |
+
+    Scenario: a user tries to prepare a project with malformed path dependency
+        When  a user generates environment for "malformed path dependency project"
+        Then  Cachi2 reports an error
+         And  repository for "malformed path dependency project" remains unchanged
+
+    Rule: fetched dependencies directory structure must replicate Bundler cache structure
+        Example: a user prepares simple Ruby project with just gem dependencies
+            When a user generates an environment for "purely gem-dependent project"
+            Then dependencies are stored in vendor/cache
+             And .gem dependencies are kept as is
+
+        Example: a user prepares simple Ruby project with git dependencies
+            When a user generates an environment for "git and gem-dependent project"
+            Then dependencies are stored in vendor/cache
+             And Git dependnecies directories  names contain revisions
+
+        Example: a user prepares complex Ruby project
+            When a user generates an environment for "complex Ruby project"
+            Then dependencies are stored in vendor/cache
+             And dependencies are not duplicated
+
+
+### Necessary repositories
+
```

Comment:
Ideally, we want to create one test repository with multiple branches. Each branch should have a descriptive name. That's our current approach to e2e / integration tests.

Reply:
Good point, thank you, will update the document to reflect that this is a single repository with multiple branches.